yarn-project/simulator/src/public/abstract_phase_manager.ts

```diff
@@ -293,23 +315,28 @@ export abstract class AbstractPhaseManager {
         if (isExecutionRequest) {
           newUnencryptedFunctionLogs.push(result.allUnencryptedLogs);
         }
+        executionStack.push(...result.nestedExecutions);
 
+        // Simulate the public kernel circuit.
         this.log.debug(
           `Running public kernel circuit for ${result.execution.contractAddress.toString()}:${functionSelector}`,
         );
-        executionStack.push(...result.nestedExecutions);
         const callData = await this.getPublicCallData(result, isExecutionRequest);
-
-        const circuitResult = await this.runKernelCircuit(kernelOutput, callData);
-        kernelOutput = circuitResult[1];
-
-        // Capture the inputs to the kernel circuit for later proving
-        publicKernelInputs.push(circuitResult[0]);
-
-        // sanity check. Note we can't expect them to just be equal, because e.g.
+        const circuitResult = await this.runKernelCircuit(kernelPublicOutput, callData);
+        kernelPublicOutput = circuitResult[1];
+
+        // Capture the inputs for later proving in the AVM and kernel.
+        const publicProvingInformation: PublicProvingInformation = {
+          calldata: result.calldata,
+          bytecode: result.bytecode!,
+          inputs: circuitResult[0],
```

Comment:
Can these also be moved to objects later, or even decomposed directly at circuitResult

Reply:
Done.